tests/BCSTests.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright © 2017-2022 Trust Wallet.
+// Created by Clément Doumergue
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "BCS.h"
+
+#include <gtest/gtest.h>
+
+namespace TW::BCS::tests {
+
+TEST(BCS, Integral) {
+    Serializer os;
+    os << uint32_t(0xAABBCCDD);
+    ASSERT_EQ(os.bytes, (Data{0xDD, 0xCC, 0xBB, 0xAA}));
```

Comment:
A general comment: Data consts are more readable if initialized with parse_hex(), especially in tests, like this:
`parse_hex("DDCCBBAA")`